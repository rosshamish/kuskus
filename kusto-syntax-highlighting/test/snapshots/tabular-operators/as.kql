// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/asoperator

// 1. In the following 2 example the union's generated TableName column will consist of 'T1' and 'T2'
range x from 1 to 10 step 1 
| as T1 
| union withsource=TableName (range x from 1 to 10 step 1 | as T2)

// Alternatively, you can write the same example, as follows:
union withsource=TableName (range x from 1 to 10 step 1 | as T1), (range x from 1 to 10 step 1 | as T2)

// 2. In the following example, the 'left side' of the join will be: 
//      MyLogTable filtered by type == "Event" and Name == "Start"
//    and the 'right side' of the join will be: 
//      MyLogTable filtered by type == "Event" and Name == "Stop"
MyLogTable  
| where type == "Event"
| as T
| where Name == "Start"
| join (
    T
    | where Name == "Stop"
) on ActivityId
