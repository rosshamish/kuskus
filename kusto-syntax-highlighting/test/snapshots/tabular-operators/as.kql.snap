>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/asoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>// 1. In the following 2 example the union's generated TableName column will consist of 'T1' and 'T2'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>range x from 1 to 10 step 1 
#^^^^^ source.kusto meta.query.range.kusto keyword.other.query.kusto
#     ^ source.kusto meta.query.range.kusto
#      ^ source.kusto meta.query.range.kusto variable.other.column.kusto
#       ^ source.kusto meta.query.range.kusto
#        ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#            ^ source.kusto meta.query.range.kusto
#             ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#              ^ source.kusto meta.query.range.kusto
#               ^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                 ^ source.kusto meta.query.range.kusto
#                  ^^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                    ^ source.kusto meta.query.range.kusto
#                     ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                         ^ source.kusto meta.query.range.kusto
#                          ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                           ^^ source.kusto
>| as T1 
#^^ source.kusto
#  ^^ source.kusto meta.query.as.kusto keyword.other.query.kusto
#    ^ source.kusto meta.query.as.kusto
#     ^^ source.kusto meta.query.as.kusto variable.other.kusto
#       ^^ source.kusto
>| union withsource=TableName (range x from 1 to 10 step 1 | as T2)
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^^^^^^^^^^^^^^^^^^^^^^^ source.kusto
#                              ^^^^^ source.kusto meta.query.range.kusto keyword.other.query.kusto
#                                   ^ source.kusto meta.query.range.kusto
#                                    ^ source.kusto meta.query.range.kusto variable.other.column.kusto
#                                     ^ source.kusto meta.query.range.kusto
#                                      ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                          ^ source.kusto meta.query.range.kusto
#                                           ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                            ^ source.kusto meta.query.range.kusto
#                                             ^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                               ^ source.kusto meta.query.range.kusto
#                                                ^^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                  ^ source.kusto meta.query.range.kusto
#                                                   ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                                       ^ source.kusto meta.query.range.kusto
#                                                        ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                         ^^^ source.kusto
#                                                            ^^ source.kusto meta.query.as.kusto keyword.other.query.kusto
#                                                              ^ source.kusto meta.query.as.kusto
#                                                               ^^ source.kusto meta.query.as.kusto variable.other.kusto
#                                                                 ^^ source.kusto
>
>// Alternatively, you can write the same example, as follows:
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>union withsource=TableName (range x from 1 to 10 step 1 | as T1), (range x from 1 to 10 step 1 | as T2)
#^^^^^ source.kusto keyword.other.query.kusto
#     ^^^^^^^^^^^^^^^^^^^^^^^ source.kusto
#                            ^^^^^ source.kusto meta.query.range.kusto keyword.other.query.kusto
#                                 ^ source.kusto meta.query.range.kusto
#                                  ^ source.kusto meta.query.range.kusto variable.other.column.kusto
#                                   ^ source.kusto meta.query.range.kusto
#                                    ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                        ^ source.kusto meta.query.range.kusto
#                                         ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                          ^ source.kusto meta.query.range.kusto
#                                           ^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                             ^ source.kusto meta.query.range.kusto
#                                              ^^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                ^ source.kusto meta.query.range.kusto
#                                                 ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                                     ^ source.kusto meta.query.range.kusto
#                                                      ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                       ^^^ source.kusto
#                                                          ^^ source.kusto meta.query.as.kusto keyword.other.query.kusto
#                                                            ^ source.kusto meta.query.as.kusto
#                                                             ^^ source.kusto meta.query.as.kusto variable.other.kusto
#                                                               ^^^^ source.kusto
#                                                                   ^^^^^ source.kusto meta.query.range.kusto keyword.other.query.kusto
#                                                                        ^ source.kusto meta.query.range.kusto
#                                                                         ^ source.kusto meta.query.range.kusto variable.other.column.kusto
#                                                                          ^ source.kusto meta.query.range.kusto
#                                                                           ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                                                               ^ source.kusto meta.query.range.kusto
#                                                                                ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                                                 ^ source.kusto meta.query.range.kusto
#                                                                                  ^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                                                                    ^ source.kusto meta.query.range.kusto
#                                                                                     ^^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                                                       ^ source.kusto meta.query.range.kusto
#                                                                                        ^^^^ source.kusto meta.query.range.kusto keyword.other.operator.kusto
#                                                                                            ^ source.kusto meta.query.range.kusto
#                                                                                             ^ source.kusto meta.query.range.kusto constant.numeric.kusto
#                                                                                              ^^^ source.kusto
#                                                                                                 ^^ source.kusto meta.query.as.kusto keyword.other.query.kusto
#                                                                                                   ^ source.kusto meta.query.as.kusto
#                                                                                                    ^^ source.kusto meta.query.as.kusto variable.other.kusto
#                                                                                                      ^^ source.kusto
>
>// 2. In the following example, the 'left side' of the join will be: 
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>//      MyLogTable filtered by type == "Event" and Name == "Start"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>//    and the 'right side' of the join will be: 
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>//      MyLogTable filtered by type == "Event" and Name == "Stop"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>MyLogTable  
#^^^^^^^^^^^^^ source.kusto
>| where type == "Event"
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^^^^^^^^^ source.kusto
#                ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                 ^^^^^ source.kusto string.quoted.double.kusto
#                      ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
>| as T
#^^ source.kusto
#  ^^ source.kusto meta.query.as.kusto keyword.other.query.kusto
#    ^ source.kusto meta.query.as.kusto
#     ^ source.kusto meta.query.as.kusto variable.other.kusto
>| where Name == "Start"
#^^ source.kusto
#  ^^^^^ source.kusto keyword.other.query.kusto
#       ^^^^^^^^^ source.kusto
#                ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                 ^^^^^ source.kusto string.quoted.double.kusto
#                      ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
>| join (
#^^ source.kusto
#  ^^^^ source.kusto keyword.other.query.kusto
#      ^^^ source.kusto
>    T
#^^^^^^ source.kusto
>    | where Name == "Stop"
#^^^^^^ source.kusto
#      ^^^^^ source.kusto keyword.other.query.kusto
#           ^^^^^^^^^ source.kusto
#                    ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                     ^^^^ source.kusto string.quoted.double.kusto
#                         ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
>) on ActivityId
#^^ source.kusto
#  ^^ source.kusto keyword.other.operator.kusto
#    ^^^^^^^^^^^^ source.kusto
>